### PR TITLE
Adds jjob and other necessities for marine LETKF task

### PR DESCRIPTION
**Pull request by AndrewEichmann-NOAA**
# Description

Adds jjob, rocoto script, config file, and other necessities for new marine LETKF task

No longer co-dependent on https://github.com/NOAA-EMC/GDASApp/pull/1090

Partially addresses https://github.com/NOAA-EMC/GDASApp/issues/1091

# Type of change

- New feature (adds functionality)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? YES
- Does this change require a documentation update? NO

# How has this been tested?

Not tested, only stubs have been added on the GDASApp side

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Comment by AndrewEichmann-NOAA** [comment]
FYI @guillaumevernieres

File: parm/config/gfs/config.resources [hunk 1]
```diff
@@ -442,6 +442,35 @@ case ${step} in
     export memory_ocnanalecen
     ;;
 
+  "ocnanalletkf")
+    npes=16
+    case ${OCNRES} in
+      "025")
+        npes=40
```
Comment (guillaumevernieres) [inline parm/config/gfs/config.resources]: might as well be realistic, start with npes=480 for the 1/4 deg case

**Review by guillaumevernieres — COMMENTED** [review]
Looks good, just 1 comment about the resources.

File: parm/config/gfs/config.resources (hunk 1 above)
Comment (AndrewEichmann-NOAA) [reply parm/config/gfs/config.resources]: done

**Comment by AndrewEichmann-NOAA** [comment]
https://github.com/NOAA-EMC/GDASApp/pull/1090 has been merged

**Review by guillaumevernieres — APPROVED** [review]

**Comment by WalterKolczynski-NOAA** [comment]
I believe this PR needs to be updated to incorporate the new GDAS hash (or wait for one of the other open workflow PRs that also update the hash).

**Comment by guillaumevernieres** [comment]
> I believe this PR needs to be updated to incorporate the new GDAS hash (or wait for one of the other open workflow PRs that also update the hash).

It's not tested in the g-w ci (for now) @WalterKolczynski-NOAA , and that task doesn't do much yet. I'd rather see this merged so @AndrewEichmann-NOAA can move on to making the bits an pieces work within our simpler testing framework.

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_LETKF [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+source "${HOMEgfs}/ush/jjob_header.sh" -e "ocnanalletkf" -c "base ocnanal ocnanalletkf"
+
+##############################################
+# Set variables used in the script
+##############################################
+# Ignore possible spelling error (nothing is misspelled)
+# shellcheck disable=SC2153
+GDATE=$(date --utc +%Y%m%d%H -d "${PDY} ${cyc} - ${assim_freq} hours")
+export gPDY=${GDATE:0:8}
+export gcyc=${GDATE:8:2}
```
Comment (aerorahul) [inline jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_LETKF]:
```suggestion
```
These are only locally used. No need to export.

File: ush/python/pygfs/task/marine_letkf.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                    chdir,
+                    Executable,
+                    FileHandler,
+                    logit,
+                    parse_j2yaml,
+                    Task,
+                    Template,
+                    TemplateConstants,
+                    WorkflowException,
+                    YAMLFile)
```
Comment (aerorahul) [inline ush/python/pygfs/task/marine_letkf.py]: Since this is a shell task, only import what is used.

File: ush/python/pygfs/task/marine_letkf.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+logger = getLogger(__name__.split('.')[-1])
+
+
+class MarineLETKF(Task):
```
Comment (aerorahul) [inline ush/python/pygfs/task/marine_letkf.py]:
```suggestion
class MarineLETKF(Analysis):
```

File: ush/python/pygfs/task/marine_letkf.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        Parameters:
+        ------------
+        None
+        Returns:
+        --------
+        None
+        """
+
+        logger.info("initialize")
+        RUN = self.runtime_config.RUN
```
Comment (aerorahul) [inline ush/python/pygfs/task/marine_letkf.py]:
```suggestion

```

File: ush/python/pygfs/task/marine_letkf.py [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        None
+        Returns:
+        --------
+        None
+        """
+
+        logger.info("finalize")
+
+        RUN = self.runtime_config.RUN
+        cyc = self.runtime_config.cyc
```
Comment (aerorahul) [inline ush/python/pygfs/task/marine_letkf.py]:
```suggestion
```

File: scripts/exgdas_global_marine_analysis_letkf.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+# Initialize root logger
+logger = Logger(level='DEBUG', colored_log=True)
+
+
+if __name__ == '__main__':
+
+    # Take configuration from environment and cast it as python dictionary
+    config = cast_strdict_as_dtypedict(os.environ)
+
+    # Instantiate the aerosol analysis task
```
Comment (aerorahul) [inline scripts/exgdas_global_marine_analysis_letkf.py]:
```suggestion
    # Instantiate the ocean LETKF task
```

File: jobs/rocoto/ocnanalletkf.sh [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+export jobid="${job}.$$"
+
+###############################################################
+# Setup Python path for GDASApp ush
+PYTHONPATH="${PYTHONPATH:+${PYTHONPATH}:}${HOMEgfs}/sorc/gdas.cd/ush"
+export PYTHONPATH
+
+###############################################################
+# Execute the JJOB
+"${HOMEgfs}"/jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_LETKF
```
Comment (aerorahul) [inline jobs/rocoto/ocnanalletkf.sh]:
```suggestion
"${HOMEgfs}/jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_LETKF"
```

File: jobs/rocoto/ocnanalletkf.sh [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+. "${HOMEgfs}/ush/load_ufsda_modules.sh"
+status=$?
+[[ ${status} -ne 0 ]] && exit "${status}"
+
+export job="ocnanalletkf"
+export jobid="${job}.$$"
+
+###############################################################
+# Setup Python path for GDASApp ush
+PYTHONPATH="${PYTHONPATH:+${PYTHONPATH}:}${HOMEgfs}/sorc/gdas.cd/ush"
```
Comment (aerorahul) [inline jobs/rocoto/ocnanalletkf.sh]:
```suggestion
PYTHONPATH="${PYTHONPATH:+${PYTHONPATH}:}${HOMEgfs}/ush/python"
```

**Review by aerorahul — CHANGES_REQUESTED** [review]

File: ush/python/pygfs/task/marine_letkf.py (hunk 3 above)
Comment (AndrewEichmann-NOAA) [reply ush/python/pygfs/task/marine_letkf.py]: I'm anticipating, but ok

**Comment by guillaumevernieres** [comment]
> I believe this PR needs to be updated to incorporate the new GDAS hash (or wait for one of the other open workflow PRs that also update the hash).

No. This pr only creates untested empty stubs and will not update the GDAS hash.

**Review by guillaumevernieres — APPROVED** [review]

File: scripts/exgdas_global_marine_analysis_letkf.py (hunk 7 above)
Comment (aerorahul) [reply scripts/exgdas_global_marine_analysis_letkf.py]: This is not an aerosol analysis task

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_LETKF [hunk 10]
```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
```
Comment (aerorahul) [inline jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_LETKF:1]: If this job is OCEAN_ANALYSIS_LETKF, then the exscript should also have the same name.
This is an EE2 detail

File: ush/python/pygfs/task/marine_letkf.py [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+from wxflow import (chdir,
+                    logit,
+                    Task)
+
+logger = getLogger(__name__.split('.')[-1])
+
+
+class MarineLETKF(Analysis):
+    """
+    Class for global ocean analysis LETKF task
```
Comment (aerorahul) [inline ush/python/pygfs/task/marine_letkf.py]: Is this ocean or marine? it might not be obvious to all what the distinctions are.

Comment (guillaumevernieres) [reply ush/python/pygfs/task/marine_letkf.py]: Good point. How about keeping marine in the class name but precising that this implementation in only for  the ocean and seaice?

Comment (AndrewEichmann-NOAA) [reply ush/python/pygfs/task/marine_letkf.py:15]: @guillaumevernieres Any preference?

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_LETKF (hunk 10 above)
Comment (AndrewEichmann-NOAA) [reply jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_LETKF:1]: @guillaumevernieres guess we got some renaming to do

Comment (aerorahul) [reply jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_LETKF:1]: I won't bother on renaming existing jobs and scripts.  Just take care of going forward so there is less to do later.

**Review by guillaumevernieres — APPROVED** [review]

**Review by aerorahul — APPROVED** [review]

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Conditionally approved pending successful completion of CI tests

**Pull request merged**